pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -734,9 +725,17 @@ func TestInsufficientCapacityNodeDaemonDoesNotUnscheduleRunningPod(t *testing.T)
 		manager.dsStore.Add(ds)
 		switch strategy.Type {
 		case apps.OnDeleteDaemonSetStrategyType:
-			syncAndValidateDaemonSets(t, manager, ds, podControl, 0, 0, 2)
+			if !utilfeature.DefaultFeatureGate.Enabled(features.ScheduleDaemonSetPods) {
+				syncAndValidateDaemonSets(t, manager, ds, podControl, 0, 0, 2)
```

Comment:
From reading these function calls, I have no idea what they are validating and whether the numbers are appropriate.

EDIT: Okay, so I went and read the code. `syncAndValidateDaemonSets` calls `validateSyncDaemonSets`. How are those two different? Well, the former is calling `manager.syncHandler(key)`, which is the only real difference. What does `syncHandler` do? :man_shrugging: I couldn't tell you, because it is defined as,
```golang
// To allow injection of syncDaemonSet for testing.
syncHandler func(dsKey string) error
```
and I don't have more time to go digging through DSController code.


Anyway, `validateSyncDaemonSets` is checking counters for create, delete, and event.